packages/protocol/contracts/layer2/based/anchor/ShastaAnchor.sol

```diff
@@ -26,6 +26,12 @@ abstract contract ShastaAnchor is PacayaAnchor {
         shastaForkHeight = _shastaForkHeight;
     }
 
+    modifier recordTxGasUsed() {
+        uint256 _gasleft = gasleft();
+        _;
+        lastAnchorGasUsed = uint32(_gasleft - gasleft());
```

Comment:
I think here we still miss the `21_000` base gas for calling this method, and all the gas for the calldata input? Not sure if this matters.

Reply:
In protocol, now we enforce the anchor tx's gas limit value to be 1000000

Reply:
I see but not sure if we need the accurate `gasUsed` value for the `anchorV4` transaction, if so, we are still missing one `SSTORE` + one `SUB` in the latest impl?

Reply:
modified the code again